Assets/Scripts/BulletManager.cs

```diff
@@ -26,8 +26,13 @@ public void SpawnBullet(Vector3 direction, Vector3 spawnPos, Vector3 rotation)
         else if (bulletType == BulletType.slow)
             bulletController = new SlowBulletController();
 
-        bulletController.SpawnBullet(direction, spawnPos, rotation);
+        return bulletController;
     }
 
+    public void RemoveController(BulletController bulletController)
+    {
+        this.bulletController = bulletController;
```

Comment:
this.bulletController is a class variable - so now you have a reference to the bulletController you are trying to remove. Since you are not storing bulletController anywhere in a list you can keep this RemoveController empty